cidc_schemas/prism/merger.py

```diff
@@ -114,7 +101,68 @@ def merge_artifact(
     if md5_hash:
         artifact_patch["md5_hash"] = md5_hash
 
-    return _update_artifact(ct, artifact_patch, artifact_uuid)
+    return _update_artifact(
+        ct, artifact_patch, artifact_uuid, uuid_path_map=uuid_path_map
+    )
+
+
+class ArtifactInfo(NamedTuple):
+    artifact_uuid: str
+    object_url: str
+    upload_type: str
+    file_size_bytes: int
+    uploaded_timestamp: str
+    crc32c_hash: Optional[str] = None
+    md5_hash: Optional[str] = None
+
+
+def merge_artifacts(
+    ct, artifacts: List[ArtifactInfo]
+) -> Tuple[dict, List[Tuple[dict, dict]]]:
+    """
+    Insert metadata for a batch of `artifacts` into `ct`, returning the modified `ct` dictionary
+    and array of 
+    """
+    # Make no modifications to `ct` if no artifacts are passed
+    if len(artifacts) == 0:
+        return ct, []
+
+    # Pre-compute the mapping from artifact UUIDs to metadata paths.
+    uuid_path_map = _get_uuid_path_map(ct)
+    merged_artifacts = [
+        merge_artifact(ct, *a, uuid_path_map=uuid_path_map)[1:] for a in artifacts
```

Comment:
no need to pass full `uuid_path_map` to the merge, as you can take only the needed entry out of it right here 
```suggestion
        merge_artifact(ct, *a, uuid_path=uuid_path_map[a. artifact_uuid])[1:] for a in artifacts
```

Reply:
Ah, nice! That's cleaner.